www/sinj04/sp2/js/index.js

```diff
@@ -0,0 +1,172 @@
+        })
+            .done((resp) => {
+                console.log(resp);
+                showLoader();
+                returnVar = JSON.parse(JSON.stringify(resp));
+                if (returnVar.status == 'ZERO_RESULTS') {
+                    errorHtml.empty();
+                    errorHtml.append($('<div></div>').text('Tato adresa nebyla nalezena: ' + address));
+                    hideLoader();
+                }
+                else {
+                    lng = returnVar.results[0].geometry.location.lng;
+                    lat = returnVar.results[0].geometry.location.lat;
+                    apiLocation = [lng, lat];
+                    callback(apiLocation);
+                }
+            })
+            .fail((resp) => {
+                console.log(resp);
+
+            })
+
+
+    }
+
+    function getRoute(start, end) {
+
+
+        var coords = [
+            SMap.Coords.fromWGS84(start[0], start[1]),
+            SMap.Coords.fromWGS84(end[0], end[1])
+        ];
+
+        // znovunačtení mapy na promazání starších routes
+        var stred = SMap.Coords.fromWGS84(longitudeWhole, latitudeWhole);
+        var mapa = new SMap(JAK.gel('mapa'), stred, 10);
+        mapa.addDefaultLayer(SMap.DEF_BASE).enable();
+        mapa.addDefaultControls();
+
+        var sync = new SMap.Control.Sync({ bottomSpace: 100 });
+        mapa.addControl(sync);
+
+        var find = function (route) {
+
+
+            var layer = new SMap.Layer.Geometry();
+            mapa.addLayer(layer).enable();
+
+            var coords = route.getResults().geometry;
+            // když se nepodaří udělat trasa
+            if (coords == undefined) {
+                hideLoader();
+                errorHtml.empty();
+                errorHtml.append($('<div></div>').text('Nepodařila se načíst trasa. Upřesněte zadané body.'));
+                info.empty();
+            }
+            else {
+                var center = mapa.computeCenterZoom(coords);
+                mapa.setCenterZoom(center[0], center[1]);
+                var geometry = new SMap.Geometry(SMap.GEOMETRY_POLYLINE, null, coords);
+                layer.addGeometry(geometry);
+                let results = route.getResults();
+                let length = Math.trunc(results.length / 1000);
+                let time = Math.trunc(results.time / 60);
+                let ascent = results.ascent;
+
+                info.empty();
+                info.append($('<div class = "length"></div>').text('Délka trasy: ' + length + ' km'));
+                info.append($('<div class = "time"></div>').text('Doba trasy: ' + time + ' minut'));
+                info.append($('<div class = "ascent"></div>').text('Celkové soupání: ' + ascent + ' m'));
+
+                hideLoader();
+            }
+        }
+
+        var route = new SMap.Route(coords, find);
+
+
+
+    }
+
+
+
+
+
+    $('button').click(function () {

```

Comment:
pridejte tomu button id, a nebo vyberte formular s tim ze pripojite na to udalost submit